code/controllers/subsystem/dbcore.dm

```diff
@@ -30,36 +27,24 @@ SUBSYSTEM_DEF(dbcore)
 
 /datum/controller/subsystem/dbcore/fire()
 	for(var/I in active_queries)
-		var/datum/DBQuery/Q = I
+		var/datum/query_result/Q = I
 		if(world.time - Q.last_activity_time > (5 MINUTES))
 			message_admins("Found undeleted query, please check the server logs and notify coders.")
 			log_sql("Undeleted query: \"[Q.sql]\" LA: [Q.last_activity] LAT: [Q.last_activity_time]")
 			qdel(Q)
 		if(MC_TICK_CHECK)
 			return
 
-/datum/controller/subsystem/dbcore/Recover()
-	connection = SSdbcore.connection
-	connectOperation = SSdbcore.connectOperation
-
 /datum/controller/subsystem/dbcore/Shutdown()
 	//This is as close as we can get to the true round end before Disconnect() without changing where it's called, defeating the reason this is a subsystem
 	if(SSdbcore.Connect())
-		var/datum/DBQuery/query_round_shutdown = SSdbcore.NewQuery("UPDATE [format_table_name("round")] SET shutdown_datetime = Now(), end_state = '[sanitizeSQL(SSticker.end_state)]' WHERE id = [GLOB.round_id]")
-		query_round_shutdown.Execute()
-		qdel(query_round_shutdown)
+		SSdbcore.Query("UPDATE [format_table_name("round")] SET shutdown_datetime = Now(), end_state = ? WHERE id = ?", list(SSticker.end_state, GLOB.round_id))
 	if(IsConnected())
 		Disconnect()
-	world.BSQL_Shutdown()
 
 //nu
 /datum/controller/subsystem/dbcore/can_vv_get(var_name)
-	return var_name != NAMEOF(src, connection) && var_name != NAMEOF(src, active_queries) && var_name != NAMEOF(src, connectOperation) && ..()

```

Comment:
You removed connection and connect operation, but not active_queries, are you sure there is no security implication with allowing admins to read this?

Because you are wrong, lists can't be read only, as you can just vv into the list and bypass all of the write checks.

Reply:
i forgot if what `active_queries` was used for, even. i'll handle this eventually.